File: C/palindrome.c
#include <stdio.h>
#include <math.h>


int reversed(int);
int main(void)
{
    int result = 0;
    for (int i = 100; i < 1000; i++)
    {
        for (int j = 100; j < 1000; j++)
        {
            int temp = i * j;
            if ((temp == reversed(temp)) && (temp > result)){
                result = temp;
                printf("%d * %d = %d\n", i, j, result);
            }
        }
    }
    printf("The last = %d\n", result);
}

int reversed(int n)
{
    int reverse = 0;
    while (n != 0)
    {
      reverse = reverse * 10;
      reverse = reverse + n%10;
      n = n/10;
    }
    return reverse;
}

File: Python/palindrome.py
#Find the largest palindrome made from the product of two 3-digit numbers.


def reverse(n):
    return int(str(n)[::-1])


def main():
    result = 0
    for i in range(100, 1000):
        for j in range(100, 1000):
            temp = i*j
            if temp == reverse(temp) and temp > result:
                result = temp
    print(result)
                

if __name__ == "__main__":
    main()